price/src/settings.rs

```diff
@@ -0,0 +1,133 @@
+    pub metrics: poc_metrics::Settings,
+    /// Tick interval (secs). Default = 60s.
+    #[serde(default = "default_interval")]
+    pub interval: i64,
+    /// Price age (get price as long as it was updated within `age` seconds of current time) (in secs).
+    /// Default = 60s.
+    #[serde(default = "default_age")]
+    pub age: u64,
+    /// Cluster Configuration
+    #[serde(default = "default_cluster")]
+    pub cluster: ClusterConfig,
+}
+
+pub fn default_source() -> String {
+    "https://api.devnet.solana.com".to_string()
+}
+
+pub fn default_log() -> String {
+    "price=debug".to_string()
+}
+
+pub fn default_interval() -> i64 {
+    60
+}
+
+pub fn default_age() -> u64 {
+    60
+}
+
+pub fn default_cluster() -> ClusterConfig {
+    ClusterConfig::default()
+}
+
+pub fn default_cache() -> String {
+    "/var/data/price".to_string()
+}
+
+impl Settings {
+    /// Load Settings from a given path. Settings are loaded from a given
+    /// optional path and can be overriden with environment variables.
+    ///
+    /// Environemnt overrides have the same name as the entries in the settings
+    /// file in uppercase and prefixed with "price_". For example
+    /// "price_LOG_" will override the log setting.
+    pub fn new<P: AsRef<Path>>(path: Option<P>) -> Result<Self, config::ConfigError> {
+        let mut builder = Config::builder();
+
+        if let Some(file) = path {
+            // Add optional settings file
+            builder = builder
+                .add_source(File::with_name(&file.as_ref().to_string_lossy()).required(false));
+        }
+        // Add in settings from the environment (with a prefix of APP)
+        // Eg.. `MI_DEBUG=1 ./target/app` would set the `debug` key
+        builder
+            .add_source(Environment::with_prefix("price").separator("_"))
+            .build()
+            .and_then(|config| config.try_deserialize())
+    }
+
+    pub fn interval(&self) -> Duration {
+        Duration::seconds(self.interval)
+    }
+
+    pub fn price_key(&self, token_type: BlockchainTokenTypeV1) -> Option<SolPubkey> {
+        match token_type {
+            BlockchainTokenTypeV1::Hnt => self
+                .cluster
+                .hnt_price_key
+                .as_ref()
+                .map(|key| SolPubkey::from_str(key).expect("unable to parse")),
+            BlockchainTokenTypeV1::Hst => self
+                .cluster
+                .hst_price_key
+                .as_ref()
+                .map(|key| SolPubkey::from_str(key).expect("unable to parse")),
+            BlockchainTokenTypeV1::Mobile => self
+                .cluster
+                .mobile_price_key
+                .as_ref()
+                .map(|key| SolPubkey::from_str(key).expect("unable to parse")),
+            BlockchainTokenTypeV1::Iot => self
+                .cluster
+                .iot_price_key
+                .as_ref()
+                .map(|key| SolPubkey::from_str(key).expect("unable to parse")),
```

Comment:
Updated with panic!, cc: @madninja cuz panic! Open to other suggestions instead of panicking but as @jeffgrunewald said we want to catch this ASAP and fix it when tokens come alive. [f8ab0b3](https://github.com/helium/oracles/pull/400/commits/f8ab0b3f81d8ddf7315cdd800beaa15b769ab9af)